Make `RowSelection::from_consecutive_ranges` public

**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

@advancedxy  noted in https://github.com/apache/datafusion/pull/10738/files#r1627775856 that keeping track of skip/scan in coe like

```rust
let row_selection = RowSelection::from(vec![
    RowSelector::skip(100),
    RowSelector::select(100),
    RowSelector::skip(150),
    RowSelector::select(50),
])
```

 is hard and it is easier to undersand and more concise to use `from_consecutive_ranges`

```rust 
let row_selection = RowSelection::from_consecutive_ranges(vec![100..200, 350..400]);
```

Sadly, this function is not public:
```
error[E0624]: associated function `from_consecutive_ranges` is private
   --> datafusion/core/src/datasource/physical_plan/parquet/access_plan.rs:43:35
    |
11  |   let row_selection = RowSelection::from_consecutive_ranges(vec![100..200, 350..400]);
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^ private associated function
    |
   ::: /Users/andrewlamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/parquet-51.0.0/src/arrow/arrow_reader/selection.rs:118:5
    |
118 | /     pub(crate) fn from_consecutive_ranges<I: Iterator<Item = Range<usize>>>(
119 | |         ranges: I,
120 | |         total_rows: usize,
121 | |     ) -> Self {
```
```

**Describe the solution you'd like**
Make `RowSelection::from_consecutive_ranges` public so it can be used by other crates 

**Describe alternatives you've considered**
<!--
A clear and concise description of any alternative solutions or features you've considered.
-->

**Additional context**
<!--
Add any other context or screenshots about the feature request here.
-->

--- a/parquet/src/arrow/arrow_reader/selection.rs
+++ b/parquet/src/arrow/arrow_reader/selection.rs
@@ -85,7 +85,7 @@ impl RowSelector {
 /// // you can also create a selection from consecutive ranges
 /// let ranges = vec![5..10, 10..15];
 /// let selection =
-///   RowSelection::from_consecutive_ranges(ranges.into_iter(), Some(20));
+///   RowSelection::from_consecutive_ranges(ranges.into_iter(), 20);
 /// let actual: Vec<RowSelector> = selection.into();
 /// assert_eq!(actual, expected);
 /// ```
@@ -118,13 +118,13 @@ impl RowSelection {
             SlicesIterator::new(filter).map(move |(start, end)| start + offset..end + offset)
         });
 
-        Self::from_consecutive_ranges(iter, Some(total_rows))
+        Self::from_consecutive_ranges(iter, total_rows)
     }
 
     /// Creates a [`RowSelection`] from an iterator of consecutive ranges to keep
     pub fn from_consecutive_ranges<I: Iterator<Item = Range<usize>>>(
         ranges: I,
-        total_rows_opt: Option<usize>,
+        total_rows: usize,
     ) -> Self {
         let mut selectors: Vec<RowSelector> = Vec::with_capacity(ranges.size_hint().0);
         let mut last_end = 0;
@@ -148,10 +148,8 @@ impl RowSelection {
             last_end = range.end;
         }
 
-        if let Some(total_rows) = total_rows_opt {
-            if last_end != total_rows {
-                selectors.push(RowSelector::skip(total_rows - last_end))
-            }
+        if last_end != total_rows {
+            selectors.push(RowSelector::skip(total_rows - last_end))
         }
 
         Self { selectors }
@@ -1147,7 +1145,7 @@ mod tests {
     #[test]
     fn test_from_ranges() {
         let ranges = [1..3, 4..6, 6..6, 8..8, 9..10];
-        let selection = RowSelection::from_consecutive_ranges(ranges.into_iter(), Some(10));
+        let selection = RowSelection::from_consecutive_ranges(ranges.into_iter(), 10);
         assert_eq!(
             selection.selectors,
             vec![
@@ -1162,23 +1160,9 @@ mod tests {
 
         let out_of_order_ranges = [1..3, 8..10, 4..7];
         let result = std::panic::catch_unwind(|| {
-            RowSelection::from_consecutive_ranges(out_of_order_ranges.into_iter(), Some(10))
+            RowSelection::from_consecutive_ranges(out_of_order_ranges.into_iter(), 10)
         });
         assert!(result.is_err());
-
-        // test without total rows specified
-        let ranges = [1..3, 4..6];
-        let selection = RowSelection::from_consecutive_ranges(ranges.into_iter(), None);
-
-        assert_eq!(
-            selection.selectors,
-            vec![
-                RowSelector::skip(1),
-                RowSelector::select(2),
-                RowSelector::skip(1),
-                RowSelector::select(2),
-            ]
-        );
     }
 
     #[test]

--- a/parquet/src/arrow/arrow_reader/selection.rs
+++ b/parquet/src/arrow/arrow_reader/selection.rs
@@ -81,6 +81,13 @@ impl RowSelector {
 ///
 /// let actual: Vec<RowSelector> = selection.into();
 /// assert_eq!(actual, expected);
+///
+/// // you can also create a selection from consecutive ranges
+/// let ranges = vec![5..10, 10..15];
+/// let selection =
+///   RowSelection::from_consecutive_ranges(ranges.into_iter(), Some(20));
+/// let actual: Vec<RowSelector> = selection.into();
+/// assert_eq!(actual, expected);
 /// ```
 ///
 /// A [`RowSelection`] maintains the following invariants:
@@ -111,13 +118,13 @@ impl RowSelection {
             SlicesIterator::new(filter).map(move |(start, end)| start + offset..end + offset)
         });
 
-        Self::from_consecutive_ranges(iter, total_rows)
+        Self::from_consecutive_ranges(iter, Some(total_rows))
     }
 
     /// Creates a [`RowSelection`] from an iterator of consecutive ranges to keep
-    pub(crate) fn from_consecutive_ranges<I: Iterator<Item = Range<usize>>>(
+    pub fn from_consecutive_ranges<I: Iterator<Item = Range<usize>>>(
         ranges: I,
-        total_rows: usize,
+        total_rows_opt: Option<usize>,
     ) -> Self {
         let mut selectors: Vec<RowSelector> = Vec::with_capacity(ranges.size_hint().0);
         let mut last_end = 0;
@@ -141,8 +148,10 @@ impl RowSelection {
             last_end = range.end;
         }
 
-        if last_end != total_rows {
-            selectors.push(RowSelector::skip(total_rows - last_end))
+        if let Some(total_rows) = total_rows_opt {
+            if last_end != total_rows {
+                selectors.push(RowSelector::skip(total_rows - last_end))
+            }
         }
 
         Self { selectors }
@@ -1136,9 +1145,9 @@ mod tests {
     }
 
     #[test]
-    fn test_empty_ranges() {
+    fn test_from_ranges() {
         let ranges = [1..3, 4..6, 6..6, 8..8, 9..10];
-        let selection = RowSelection::from_consecutive_ranges(ranges.into_iter(), 10);
+        let selection = RowSelection::from_consecutive_ranges(ranges.into_iter(), Some(10));
         assert_eq!(
             selection.selectors,
             vec![
@@ -1149,7 +1158,27 @@ mod tests {
                 RowSelector::skip(3),
                 RowSelector::select(1)
             ]
-        )
+        );
+
+        let out_of_order_ranges = [1..3, 8..10, 4..7];
+        let result = std::panic::catch_unwind(|| {
+            RowSelection::from_consecutive_ranges(out_of_order_ranges.into_iter(), Some(10))
+        });
+        assert!(result.is_err());
+
+        // test without total rows specified
+        let ranges = [1..3, 4..6];
+        let selection = RowSelection::from_consecutive_ranges(ranges.into_iter(), None);
+
+        assert_eq!(
+            selection.selectors,
+            vec![
+                RowSelector::skip(1),
+                RowSelector::select(2),
+                RowSelector::skip(1),
+                RowSelector::select(2),
+            ]
+        );
     }
 
     #[test]

